Source/DafnyLanguageServer/Language/Symbols/SymbolTableFactory.cs

```diff
@@ -335,11 +386,12 @@ private class SymbolDeclarationLocationVisitor : ISymbolVisitor<Unit> {
 
       public Unit Visit(ScopeSymbol scopeSymbol) {
         cancellationToken.ThrowIfCancellationRequested();
+        var endToken = scopeSymbol.BodyEndToken;
         RegisterLocation(
           scopeSymbol,
-          scopeSymbol.Declaration.Tok,
-          scopeSymbol.Declaration.Tok.GetLspRange(),
-          new Range(scopeSymbol.Declaration.Tok.GetLspPosition(), scopeSymbol.Declaration.EndTok.GetLspPosition())
+          scopeSymbol.BodyStartToken,
+          scopeSymbol.BodyStartToken.GetLspRange(),
+          new Range(scopeSymbol.BodyStartToken.GetLspPosition(), endToken.GetLspPosition())
```

Comment:
Don't you want `ToLspPosition(token.line, token.col + token.val.Length)` here? (as in GetLspRange?)

Reply:
(I've added details about this above; copied here for context:)

> It's unclear to me what the semantics of end tokens are. For example when you convert to a range you use GetLSPPosition on both tokens, but that means the range ends on the beginning of the last token. If the last token is part of the range, the range should end at the end of that token instead.